scripts/builtin/residencyMatchMain.dml

```diff
@@ -0,0 +1,194 @@
+#                                               i.e. the leftmost value in a row in P is the preference value for the acceptor with index 1 and vice-versa (higher is better).
+# OUTPUT PARAMETERS:
+# --------------------------------------------------------------------------------------------
+# NAME                  TYPE       DEFAULT      MEANING
+# --------------------------------------------------------------------------------------------
+# ResidencyMatch        Matrix     ---          Result Matrix
+#                                               If cell [i,j] is non-zero, it means that Resident i has matched with Hospital j.
+#                                               Further, if cell [i,j] is non-zero, it holds the preference value that led to the match.
+#
+#
+# HospitalMatch         Matrix     ---          Result Matrix
+#                                               If cell [i,j] is non-zero, it means that Resident i has matched with Hospital j.
+#                                               Further, if cell [i,j] is non-zero, it holds the preference value that led to the match.
+#
+#
+# Residents.mtx:
+# 2.0,1.0,3.0
+# 1.0,2.0,3.0
+# 1.0,2.0,0.0
+#
+# Since it is an ORDERED  matrix, this means that Resident 1 (row 1) likes acceptor 2 the most, followed by acceptor 1 and acceptor 3.
+# If it was UNORDERED, this would mean that proposer 1 (row 1) likes acceptor 3 the most (since the value at [1,3] is the row max),
+# followed by acceptor 1 (2.0 preference value) and acceptor 2 (1.0 preference value).
+#
+# Hospitals.mtx:
+# 2.0,1.0,0.0
+# 0.0,1.0,2.0
+# 1.0,2.0,0.0
+#
+# Since it is an UNORDERED matrix this means that Hospital 1 (row 1) likes Resident 1 the most (since the value at [1,1] is the row max).
+#
+# Capacity.mtx
+# 1.0
+# 1.0
+# 1.0
+#
+# ResidencyMatch.mtx
+# 0.0,0.0,3.0
+# 1.0,0.0,0.0
+# 0.0,2.0,0.0
+#
+# HospitalMatch.mtx
+# 0.0,1.0,0.0
+# 0.0,0.0,2.0
+# 1.0,0.0,0.0
+#
+# Resident 1 has matched with Hospital 3 (since [1,3] is non-zero) at a preference level of 3.0.
+# Resident 2 has matched with Hospital 1 (since [2,1] is non-zero) at a preference level of 1.0.
+# Resident 3 has matched with Hospital 2 (since [3,2] is non-zero) at a preference level of 2.0.
+# --------------------------------------------------------------------------------------------
+m_residencyMatchMain = function(Matrix[Double] R, Matrix[Double] H,Matrix[Double] C )
+  return (Matrix[Double] ResidencyMatch)
+{
+#TODO PLEASE PAY ATTENTION
+# in this step we consider that  Residents Matrix is ORDERED   !!!!
+# in this step we consider that  Hospital  Matrix is UNORDERED !!!!
+
+print("STARTING Resident Hospital Match");
+#TODO set a finite number of maximum iterations so that the execution termiates after maximum iterations.
+
+print("\n")
+print("STARTING RESIDENCY MATCH ALGORITHM");
+print("READING R  as residents AND H as Hospitals and also C as capacity...");
+
+m = nrow(R)
+n = ncol (R)
+Capacityrows = nrow(C)
```

Comment:
just to follow the consistent naming conventions please start the variable name with a lower-case.

Reply:
done